spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1314,6 +1314,10 @@ def tab_name_editor(self):
     @Slot(object, object)
     def edit_file(self, filename, line):
         """Handle %edit magic petitions."""
+        if not osp.isfile(filename):
+            # Create new file
+            with open(filename, "w") as f:
+                f.write("")
```

Comment:
I think we should add our template for untitled files here, instead of creating the file with empty content. What do you think?